### PR TITLE
feat(db,api): filter workers in db, not in browser

### PR DESCRIPTION
**Pull request by matt-boris**
Addresses #5456.

> This change adds more DB functions to allow for filtering in the DB based on worker state and quarantined status for the workers page. Previously, filtering would only happen on the initial page loaded from the DB if results were paginated. This should also speed up the workers page rendering when a filter is applied.

### CODE REVIEW

File: services/worker-manager/src/api.js [hunk 1]
```diff
@@ -868,10 +868,7 @@ builder.declare({
     'page. You may limit this with the query-string parameter `limit`.',
   ].join('\n'),
 }, async function(req, res) {
-  const quarantined = req.query.quarantined || null;
```
Comment (lotas) [inline services/worker-manager/src/api.js:871]: those are listed in the api definition above on line 848-849, if they are no longer used, should be dropped from declaration.
Are they still used?

Comment (lotas) [reply services/worker-manager/src/api.js:871]: ah, full query is passed I see, nevermind

File: services/worker-manager/src/data.js [hunk 2]
```diff
@@ -337,13 +337,32 @@ class Worker {
     const fetchResults = async (query) => {
       const { continuationToken, rows } = await paginateResults({
         query,
-        fetch: (size, offset) =>
-          db.fns.get_queue_workers_with_wm_join(
+        fetch: (size, offset) => {
+          if (Object.keys(query).includes('workerState')) {
```
Comment (lotas) [inline services/worker-manager/src/data.js:341]: we should probably put this into endpoint description that you can only use one at a time to filter value: either state or qurantined

**Review by lotas — APPROVED** [review]

File: services/worker-manager/src/data.js (hunk 2 above)
Comment (matt-boris) [reply services/worker-manager/src/data.js:341]: Ah true, good call. Thanks. I was thinking of only _our_ use case which is directly through the UI where only one filter is selectable in the first place. Will update.

Comment (matt-boris) [reply services/worker-manager/src/data.js:341]: It's actually already in the docs :) https://firefox-ci-tc.services.mozilla.com/docs/reference/core/worker-manager/api#listWorkers

**Pull request merged**